From 8d1c1437a9a3ccebcf575b1f92811bdff008c09c Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Mon, 21 Sep 2020 13:22:54 +0200
Subject: [PATCH 808/813] neighbor_db: set the co-locate bit

If the BSS is co-located, the corresponding bit needs to be set inside
bssid_info.

Signed-off-by: John Crispin <john@phrozen.org>
---
 src/ap/neighbor_db.c         | 3 +++
 src/common/ieee802_11_defs.h | 2 ++
 2 files changed, 5 insertions(+)

diff --git a/src/ap/neighbor_db.c b/src/ap/neighbor_db.c
index 01bf88623..3a4cdc004 100644
--- a/src/ap/neighbor_db.c
+++ b/src/ap/neighbor_db.c
@@ -260,6 +260,9 @@ void hostapd_neighbor_set_own_report(struct hostapd_data *hapd)
 			bssid_info |= NEI_REP_BSSID_INFO_HE;
 	}
 
+	if (hapd->iconf->he_co_locate)
+		bssid_info |= NEI_REP_BSSID_INFO_HE_CO_LOCATED;
+
 	/* TODO: Set NEI_REP_BSSID_INFO_MOBILITY_DOMAIN if MDE is set */
 
 	if (ieee80211_freq_to_channel_ext(hapd->iface->freq,
diff --git a/src/common/ieee802_11_defs.h b/src/common/ieee802_11_defs.h
index ec3a0613f..6c0eef885 100644
--- a/src/common/ieee802_11_defs.h
+++ b/src/common/ieee802_11_defs.h
@@ -2130,6 +2130,8 @@ enum phy_type {
 #define NEI_REP_BSSID_INFO_VHT BIT(12)
 #define NEI_REP_BSSID_INFO_FTM BIT(13)
 #define NEI_REP_BSSID_INFO_HE BIT(14)
+#define NEI_REP_BSSID_INFO_HE_ER_BSS BIT(15)
+#define NEI_REP_BSSID_INFO_HE_CO_LOCATED BIT(16)
 
 /*
  * IEEE P802.11-REVmc/D5.0 Table 9-152 - HT/VHT Operation Information
-- 
2.25.1

